src/nextjournal/clerk/viewer.cljc

```diff
@@ -387,11 +387,15 @@
      (str/join (repeat (get (frequencies (str path)) \/ 0) "../"))))
 
 #?(:clj
-   (defn maybe-store-result-as-file [{:as doc+blob-opts :keys [out-path file]} {:as result :nextjournal/keys [content-type value]}]
+   (defn map-index [{:as _opts :keys [index]} path]
+     (get {index "index.clj"} path path)))
+
+#?(:clj
+   (defn maybe-store-result-as-file [{:as doc+blob-opts :keys [index out-path file]} {:as result :nextjournal/keys [content-type value]}]
      ;; TODO: support customization via viewer api
      (if-let [image-type (second (re-matches #"image/(\w+)" content-type))]
        (assoc result :nextjournal/value
-              (str (relative-root-prefix-from file)
+              (str (relative-root-prefix-from (map-index doc+blob-opts file))
```

Comment:
Slightly prefer this to the second index arg to `relative-root-prefix-from` as we can later grow map-index into a general routing function.